fseai/vision.py

```diff
@@ -0,0 +1,129 @@
+"""
+import torch
+from torchvision import transforms, datasets
+from torch.utils.data import DataLoader
+import torch.nn as nn
+from torch import optim
+from FSEai import fseai_models as models
+
+
+def input_mnist():
+    """
+    Function to load mnist dataset
+    """
+    trainset = datasets.MNIST('', download=True, train=True, transform=transforms.ToTensor())
+    testset = datasets.MNIST('', download=True, train=False, transform=transforms.ToTensor())
+
+    print('MNIST train and test data set Loaded!')
+    return trainset, testset
+
+
+def build_mnist_model():
+    """
+    Function to build a fully connected model 
+    :return: model
+    """
+    # Import the Fully Connected network model from fseai_models
+    model = models.FCmodel
+    input_size = 784
+    hidden_size = [128, 64]
+    output_size = 10
+    model = model(input_size, hidden_size, output_size)
+    return model
+
+
+def train_mnist(model, trainset):
+    """
+    Function to train mnist data with Fully Connected model
+    """
+    global correct, total
+    train_loader = DataLoader(dataset=trainset, batch_size=64, shuffle=True)
+    # Loss and optimizer
+    lossFunction = nn.NLLLoss()
+    optimizer = optim.SGD(model.parameters(), lr=0.003, momentum=0.9)
+    num_epochs = 10
+    loss_mnist = []
+    accuracy_mnist = []
+    print('Training ......')
+    for epoch in range(num_epochs):
+        for i, (images, labels) in enumerate(train_loader):
+            # Flatten the input images of [28,28] to [1,784]
+            images = images.reshape(-1, 784)
+
+            # Forward Pass
+            outputs = model(images)
+            loss = lossFunction(outputs, labels)
+            loss_mnist.append(loss.item())
+
+            # Backprop and perform Adam optimisation
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+
+            # Track the accuracy
+            total = labels.size(0)
+            _, predicted = torch.max(outputs.data, 1)
+            correct = (predicted == labels).sum().item()
+            accuracy_mnist.append(correct/total)
+
+        print('Epoch %s, Training Accuracy: %s' % (epoch, (correct/total) * 100))
+    print('Model training with MNIST train data set completed!')
+    return model
+
+
+def test_mnist(model, testset):
+    """
+    Test the cnn model for mnist test dataset
+    """
+    test_loader = DataLoader(dataset=testset, batch_size=64, shuffle=False)
+    model.eval()
+    with torch.no_grad():
+        correctly_classified = 0
+        total_classified = 0
+        for images, labels in test_loader:
+            images = images.reshape(-1, 784)
+            outputs = model(images)
+            _, predicted = torch.max(outputs.data, 1)
+            total_classified += labels.size(0)
+            correctly_classified += (predicted == labels).sum().item()
+
+    print('Test Accuracy of the model on the test dataset: %s' % ((correctly_classified / total_classified) * 100))
```

Comment:
I think the line length is more than 90. While pep8's recommendation is 79, I think we should stick with 90